### ISSUE TITLE
Generated Code in VS2019 Do Not Pass Analyze

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#45423]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/1087980/generated-code-in-vs2019-do-not-pass-analyze.html)._

---
For .NET Core projects, it seems files generated by VS causes violations after running Analyze Solution.

For example the AssemblyAttributes.cs (auto-generated) that causes an error like this:
`IDE0005 Using directive is unnecessary.` 
Another example would be for code like this:

```
if (env.IsDevelopment())
{
    app.UseDeveloperExceptionPage();
}
```

This creates the error "`Expression Value is Never Used`" message.

---
#### Visual Studio Feedback System on 6/23/2020, 00:16 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 6/24/2020, 09:51 AM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by sharwell** [comment]
> For example the AssemblyAttributes.cs (auto-generated) that causes an error like this:
> `IDE0005 Using directive is unnecessary.`

IDE0005 is not reported in generated code. We would need to see a specific example of a project that produces this warning for two reasons:

1. We need to see what code or extension is generating the AssemblyAttributes.cs
2. We need to see why the generated file is not being interpreted as generated code

**Comment by bordecal** [comment]
@sharwell I am seeing this problem as well (VS 16.6.5).  Repro steps:

1. Enable [Full Solution Analysis](https://docs.microsoft.com/en-us/visualstudio/code-quality/configure-live-code-analysis-scope-managed-code?view=vs-2019) in Visual Studio.
2.  Adjust root `.editorconfig` to default to the `error` severity level for all rules by adding `dotnet_analyzer_diagnostic.severity = error`.
3. Compile pretty much any type of C# project (e.g.: .NET Standard or full Fx class library).  (sample solution: [GeneratedObjFiles.zip](https://github.com/dotnet/roslyn/files/5001412/GeneratedObjFiles.zip))


Two errors (`IDE0005_gen` and `RemoveUnnecessaryImportsFixable`) will be observed for each of the `______________.AssemblyAttributes.cs` and `______________.AssemblyInfo.cs` generated files in the `obj` folder.

Both of the generated files contain an `auto-generated` header and should presumably be [excluded from analysis based on this](https://docs.microsoft.com/en-us/visualstudio/code-quality/use-roslyn-analyzers?view=vs-2019#configure-generated-code).  Unfortunately, this doesn't seem to be the case,  I've tried identifying the files as generated code using the `generated_code = true` setting in `.editorconfig` but haven't been able to find any file path pattern that actually works.

Luckily, it appears to be possible to completely disable these two diagnostics (e.g.: `dotnet_diagnostic.IDE0005_gen.severity = none` in `.editorconfig`) without affecting the quality of analysis results since `IDE0005` will still generate violations for non-generated files even in the absence of those rules.  However, it is still potentially problematic that violations are being surfaced at all from these generated files since it could cause issues with other rules that users might not be willing to completely disable.

(FWIW, I'm personally hoping to be able to forgo full solution analysis once the CodeStyle analyzer package is released.  I suspect that might also be the case for many/most folks who would choose to apply `dotnet_analyzer_diagnostic.severity = error`.  However, the feasibility of dropping full solution analysis will depend very much on whether all "interesting" rules are covered by CodeStyle.)

**Comment by sharwell** [comment]
Duplicate of #43305

**Comment by sharwell** [comment]
@bordecal Thanks for the additional info. Under the design, `IDE0005_gen` is not configurable, which means it's not allowed to change it to error. Unfortunately, a bug in one of our configuration options bypassed this protection (see #43305) which led to the behavior you observed. The issue has been fixed for 16.7 starting with Preview 1.
